yasnippet.el

```diff
@@ -3343,20 +3343,24 @@ This renders the snippet as ordinary text."
                    (if (symbolp fun) fun "a hook")
                    (error-message-string error)))))
 
-(defun yas--safely-run-hook (hook)
-  (let ((debug-on-error (and (not (memq yas-good-grace '(t hooks)))
+(defun yas--safely-run-hook (hook-symbol)
+  (let ((hook (symbol-value hook-symbol))
+        (debug-on-error (and (not (memq yas-good-grace '(t hooks)))
                              debug-on-error)))
     (if (functionp hook) (yas--safely-call-fun hook)
-      (mapc #'yas--safely-call-fun hook))))
+      (if (not (member t hook))
+          (mapc #'yas--safely-call-fun hook)
+        (setq hook (remove t hook))
+        (mapc #'yas--safely-call-fun hook)
+        (mapc #'yas--safely-call-fun (default-value hook-symbol))))))
```

Comment:
I would prefer to run the `default-value` based on where `t` is located in the list, rather than unconditionally doing it at the end. Although it's true `add-hook` always leaves `t` at the end of the list so we can probably get away with this.